c7n/resources/asg.py

```diff
@@ -988,6 +988,49 @@ def process_asg(self, asg):
             raise
 
 
+@actions.register('modify-security-groups')
+class ASGModifyGroups(ModifyGroupsAction):
+    # Need some kind of a warning - this will require the ASGs to bring up new instances
+    # in order to see the new security groups
+
+    # TODO: Implement. Lots of other gotchas
+
+    def process(self, asgs):
+        raise NotImplemented('The modify-security-groups action has not been implemented on this resource')
+        with self.executor_factory(max_workers=5) as w:
+            list(w.map(self.process_asg, asgs))
+
+    def process_asg(self, asg):
+        client = local_session(self.manager.session_factory).client('autoscaling')
+
+        # patch the resource obj for the generic actions
+        asg['Groups'] = asg['SecurityGroups']
```

Comment:
better to make a base method `get_group_ids(resource)` that this could subclass and leave the generics out of annotations, from an annotation perspective this is straight copying of keys, resource records need to be indexed and straight key copying just adds weight.